modules/telestion-application/src/main/java/de/wuespace/telestion/application/loader/Loader.java

```diff
@@ -0,0 +1,350 @@
+	 * an error during execution.
+	 * <p>
+	 * The launcher calls this event handler. <strong>Don't call it by yourself.</strong>
+	 * <p>
+	 * <em>Call chain</em>:
+	 * <strong><code>onBeforeVertxShutdown</code></strong> ➔
+	 * {@link #onAfterVertxShutdown(Promise) onAfterVertxShutdown} ➔
+	 * {@link #onExit(Promise) onExit}
+	 *
+	 * @param stopPromise the promise that indicates the completion state of this loader
+	 */
+	void onBeforeVertxShutdown(Promise<Void> stopPromise) throws Exception;
+
+	/**
+	 * Gets called, after the {@link ConfigDeploymentLauncher Launcher} closes the
+	 * {@link io.vertx.core.Vertx Vertx} instance, but before it removes the instance and the related
+	 * {@link io.vertx.core.VertxOptions VertxOptions}.
+	 * <p>
+	 * Use this event if you want to run shutdown steps that aren't depend on a running Vertx instance but running
+	 * before the exit step.
+	 * <p>
+	 * The passed promise should resolve when this loader finishes its execution and fail if the loader encounter
+	 * an error during execution.
+	 * <p>
+	 * The launcher calls this event handler. <strong>Don't call it by yourself.</strong>
+	 * <p>
+	 * <em>Call chain</em>:
+	 * {@link #onBeforeVertxShutdown(Promise) onBeforeVertxShutdown} ➔
+	 * <strong><code>onAfterVertxShutdown</code></strong> ➔
+	 * {@link #onExit(Promise) onExit}
+	 *
+	 * @param stopPromise the promise that indicates the completion state of this loader
+	 */
+	void onAfterVertxShutdown(Promise<Void> stopPromise) throws Exception;
+
+	/**
+	 * Gets called, after the {@link ConfigDeploymentLauncher Launcher} removes the {@link io.vertx.core.Vertx Vertx}
+	 * instance and {@link io.vertx.core.VertxOptions VertxOptions} and is ready to complete the shutdown process.
+	 * <p>
+	 * Use this event if you want to run shutdown steps that should run directly before the launcher stops.
+	 * <p>
+	 * The passed promise should resolve when this loader finishes its execution and fail if the loader encounter
+	 * an error during execution.
+	 * <p>
+	 * The launcher calls this event handler. <strong>Don't call it by yourself.</strong>
+	 * <p>
+	 * <em>Call chain</em>:
+	 * {@link #onBeforeVertxShutdown(Promise) onBeforeVertxShutdown} ➔
+	 * {@link #onAfterVertxShutdown(Promise) onAfterVertxShutdown} ➔
+	 * <strong><code>onExit</code></strong>
+	 *
+	 * @param stopPromise the promise that indicates the completion state of this loader
+	 */
+	void onExit(Promise<Void> stopPromise) throws Exception;
+
+	/**
+	 * Gets called, before the {@link ConfigDeploymentLauncher Launcher} deploys a new
+	 * {@link io.vertx.core.Verticle Verticle}.
+	 * <p>
+	 * Use this event if you want to run steps before a verticle deploys.
+	 * <p>
+	 * The passed promise should resolve when this loader finishes its execution and fail if the loader encounter
+	 * an error during execution. A failed promise <strong>prevents</strong> the verticle from deploying.
```

Comment:
Do the other Verticles continue to get deployed (only skipping this one Verticle) or is the whole process stopped? Please specify

Reply:
Only this specific deployment doesn't get deployed.

Reply:
Done in [`46e0777` (#501)](https://github.com/wuespace/telestion-core/pull/501/commits/46e07771cc2222f313d84472f69aa37a80b05033). Please re-review if you don't like it. :wink: